Shared/Samples/Authenticate with OAuth/AuthenticateWithOAuthView.swift

```diff
@@ -0,0 +1,96 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+import ArcGIS
+import ArcGISToolkit
+
+struct AuthenticateWithOAuthView: View {
+    /// The authenticator to handle authentication challenges.
+    @StateObject private var authenticator = Authenticator(
+        oAuthUserConfigurations: [.arcgisDotCom]
+    )
+
+    init() {
+        // Sets authenticator as ArcGIS and Network challenge handlers to handle authentication
+        // challenges.
+        ArcGISEnvironment.authenticationManager.handleChallenges(using: authenticator)
```

Comment:
Not a big deal, but this line would trigger the following warning in console

```
[SwiftUI] Accessing StateObject's object without being installed on a View.
This will create a new instance each time.
```

![image](https://user-images.githubusercontent.com/9660181/226492004-6667ae9f-8cfb-422e-b480-8d669400524a.png)

It can be fixed by moving the logic into `onAppear` as [here](https://github.com/Esri/arcgis-maps-sdk-swift-samples/pull/126#discussion_r1140793697), which I think still make sense in terms of a view lifecycle.

Reply:
Fixed. There are consequences of doing this in `onAppear` and `onDisappear`. In this sample there is only one view so it should be ok.

Reply:
Thanks for adding the detailed comments!